File: app/src/main/java/com/mya/games/colipop/board/Cell.java
package com.mya.games.colipop.board;

public class Cell {
    // Board cell position
    public int posX = 0;
    public int posY = 0;

    // Flag to ignore cell logic
    public boolean ignore = false;

    public Bubble bubble = null;
    public Thing thing = null;
    public Explosion explosion = null;

    public Cell(int posX, int posY) {
        this.posX = posX;
        this.posY = posY;
    }
}

File: app/src/main/java/com/mya/games/colipop/board/Cell.kt
package com.mya.games.colipop.board

class Cell(posX: Int, posY: Int) {
    // Board cell position
    var posX = 0
    var posY = 0

    // Flag to ignore cell logic
    var ignore = false

    var bubble: Bubble? = null
    var thing: Thing? = null
    var explosion: Explosion? = null

    init {
        this.posX = posX
        this.posY = posY
    }
}
